src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/AssociatedCaseLinkHelper.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
+import uk.gov.hmcts.reform.sscs.ccd.service.CcdService;
+import uk.gov.hmcts.reform.sscs.idam.IdamService;
+import uk.gov.hmcts.reform.sscs.idam.IdamTokens;
+
+@Component
+@Slf4j
+public class AssociatedCaseLinkHelper {
+
+    private final CcdService ccdService;
+    private final IdamService idamService;
+
+    @Autowired
+    public AssociatedCaseLinkHelper(CcdService ccdService, IdamService idamService) {
+        this.ccdService = ccdService;
+        this.idamService = idamService;
+    }
+
+    public void linkCaseByNino(SscsCaseData sscsCaseData) {
```

Comment:
could this return the updated SSCSCaseData to make it purer?

Reply:
updated